Mage.Server.Plugins/Mage.Player.Human/src/mage/player/human/HumanPlayer.java

```diff
@@ -617,33 +622,40 @@ public boolean chooseTarget(Outcome outcome, Target target, Ability source, Game
         Map<String, Serializable> options = new HashMap<>();
 
         while (canRespond()) {
-            Set<UUID> possibleTargets = target.possibleTargets(abilityControllerId, source, game);
+            Set<UUID> possibleTargetIds = target.possibleTargets(abilityControllerId, source, game);
             boolean required = target.isRequired(source != null ? source.getSourceId() : null, game);
-            if (possibleTargets.isEmpty()
+            if (possibleTargetIds.isEmpty()
                     || target.getTargets().size() >= target.getNumberOfTargets()) {
                 required = false;
             }
 
-            java.util.List<UUID> chosen = target.getTargets();
-            options.put("chosen", (Serializable) chosen);
+            UUID responseId = target.tryToAutoChoose(abilityControllerId, source, game);
 
-            updateGameStatePriority("chooseTarget", game);
-            prepareForResponse(game);
-            if (!isExecutingMacro()) {
-                game.fireSelectTargetEvent(getId(), new MessageToClient(target.getMessage(), getRelatedObjectName(source, game)),
-                        possibleTargets, required, getOptions(target, options));
+            // responseId is null if a choice couldn't be automatically made
+            if (responseId == null) {
+
+
+                List<UUID> chosen = target.getTargets();
+                options.put("chosen", (Serializable) chosen);
```

Comment:
1. You must check your changes again with option names. I can see errors with it (wrong/miss replace). As I can see replaces must be:
- `chosen` -> `chosenTargets`
- `choosable` -> possibleTargets`

2. Also you must change client side code too for work with new options:

![shot_220703_003728](https://user-images.githubusercontent.com/8344157/177015607-36889365-2af2-4168-83e5-ad1f75da540d.png)

Reply:
I think my most recent push should address this.